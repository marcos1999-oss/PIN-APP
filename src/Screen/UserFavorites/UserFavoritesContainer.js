import React from "react";
import { Animated, FlatList, View, Image, TouchableOpacity, Platform } from "react-native";
import {

  Text,
  Container,
  Header,
  Left,
  Body,
  Right,
  Button,
  Content,
} from "native-base";
import { SwipeRow } from "react-native-swipe-list-view";
import CustomIcon from "../../CustomIcon";
import { errorMessage } from '../../Utils/alerts'

import styles from "./style";
import { scale, verticalScale } from "../../Utils/scaling";
import { CachedImage } from 'react-native-cached-images';

//Redux
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import {
  getLikedPostsPagination,
} from '../../redux/selectors/index';
import * as postActions from '../../redux/actions/postActions';


export const DealItem = ({ item, onPressItem }) => {
  const subText = `${item.firm.city} ${item.distance.toString()} mi`;

  return (
    <TouchableOpacity
      activeOpacity={1}
      style={styles.itemContainer}
      onPress={() => onPressItem()}
    >
      <View style={styles.branchContainer}>
        <View style={styles.branchImageContainer}>
          <CachedImage source={{ uri: item.firm.photo.url }} style={styles.branchImage} />
        </View>

        <Text style={styles.branchName} numberOfLines={1}>
          {item.firm.name}
        </Text>
      </View>

      <View style={styles.dealInfo}>
        <Text style={styles.title} numberOfLines={1}>
          {item.title}
        </Text>

        <Text style={styles.subText}>{subText}</Text>
      </View>
    </TouchableOpacity>
  );
};


class UserFavoriteComponent extends React.Component {

  goToAdView = (...e) => {
    this.props.navigation.navigate("UserAdView");
  };

  _renderItem = ({ item }) => {
    return (
      <View style={styles.swipeContainer}>
        <SwipeRow
          closeOnScroll={true}
          previewRowKey={item.id}
          rightOpenValue={scale(-75)}
          disableRightSwipe
          closeOnRowPress={false}
          onSwipeValueChange={(s) => {
            /*

            if (s.direction === "right") {
              this.removeItem(item);
            }*/
          }}
        >
          <Button
            transparent
            onPress={() => this.removeItem(item)}
            style={styles.deleteBtn}
          >
            <Text style={styles.deleteText}>Delete</Text>
          </Button>

          <DealItem item={item} onPressItem={this.goToAdView} />
        </SwipeRow>
      </View>
    );
  };

  removeItem = (post) => {
    this.props.postActions.unfavoritePost({
      post,
      onSuccess: () => {

      },
      onFail: (error) => {
        errorMessage({ message: 'Could not remove this post from favorites', description: error.message });
      }
    });
  };

  onListEndReached() {
    if (this.props.pagination.currentPage !== this.props.pagination.totalPages) {
      this.props.postActions.fetchLikedPosts({
        currentPage: this.props.pagination.currentPage + 1,
        perPage: this.props.pagination.perPage
      });
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <Header style={styles.header}>
          <Left style={{ flex: 1 }} />
          <Body>
            <Text style={styles.headerContentStyle}>Favorites</Text>
          </Body>
          <Right>
            <Button transparent onPress={() => this.props.navigation.goBack()}>
              <CustomIcon name="close" style={styles.headerContentStyle} />
            </Button>
          </Right>
        </Header>
        <View style={{ flex: 1 }}>
          <FlatList
            data={this.props.likedPosts}
            style={{ width: '100%' }}
            contentContainerStyle={styles.favoritesContent}
            renderItem={this._renderItem}
            keyExtractor={(item, idx) => item.id}
            getItemLayout={(data, index) => (
              { length: verticalScale(56.44), offset: verticalScale(56.44) * index, index }
            )}
            onEndReachedThreshold={0.2}
            onEndReached={() => this.onListEndReached()}
          />
        </View>
      </View>
    );
  }
}

const mapStateToProps = state => ({
  pagination: getLikedPostsPagination(state),
});

const mapDispatchToProps = dispatch => ({
  postActions: bindActionCreators(postActions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(UserFavoriteComponent)
