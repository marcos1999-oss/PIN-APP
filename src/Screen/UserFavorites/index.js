import React from "react";
import { NavigationEvents } from "react-navigation";
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { getPostsMeta, getLikedPosts, getLikedPostsPagination } from '../../redux/selectors/index'
import * as postActions from "../../redux/actions/postActions"

import UserFavoriteComponent from "./UserFavoritesContainer";


class UserFavorites extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      loading: false,
    };
  }

  componentDidMount() {
    this.props.postActions.fetchLikedPosts({
      currentPage: this.props.pagination.currentPage,
      perPage: this.props.pagination.perPage
    });
  }

  render() {
    return (
      <React.Fragment>
        {!this.props.isTesting && (<NavigationEvents
          onDidFocus={this.props.navigation.getParam('onDidFocus')}
        />)}

        <UserFavoriteComponent {...this.props} />
      </React.Fragment>
    );
  }
}

const mapStateToProps = state => {
  return {
    postsMeta: getPostsMeta(state),
    likedPosts: getLikedPosts(state),
    pagination: getLikedPostsPagination(state),
  }
};

const mapDispatchToProps = dispatch => {
  return {
    postActions: bindActionCreators(postActions, dispatch),
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(UserFavorites);
