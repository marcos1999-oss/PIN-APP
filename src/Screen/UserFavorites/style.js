import { StyleSheet } from "react-native";
import {
  verticalScale,
  scale,
  normalizeSize,
  moderateScale,
} from "../../Utils/scaling";
import { responsiveHeight } from "../../Utils/dimensions";

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#5F92F3",
  },
  header: {
    backgroundColor: "#5F92F3",
    borderBottomWidth: 0,
    marginTop: responsiveHeight(2),
    elevation: 0,
  },
  headerContentStyle: {
    color: "white",
    fontSize: moderateScale(12),
    fontWeight: "bold",
    alignSelf: "center",
  },
  closeIcon: {
    color: "white",
  },
  favoritesContent: {
    marginLeft: scale(20),
    marginRight: scale(15),
    flexDirection: "column",
    justifyContent: "flex-start",
  },
  swipeContainer: {
    backgroundColor: "#00000000",
    height: verticalScale(56.44),
    marginBottom: verticalScale(20),
    paddingTop: 0,
    paddingRight: 0,
    paddingBottom: 0,
    borderBottomWidth: 0,
    borderRadius: verticalScale(20),
  },
  deleteBtn: {
    height: verticalScale(56.44),
    marginBottom: verticalScale(20),
    alignSelf: "flex-end",
  },
  deleteText: {
    fontSize: moderateScale(10),
    fontWeight: "bold",
    color: "white",
  },
  itemContainer: {
    height: verticalScale(56.44),
    paddingVertical: verticalScale(9),
    paddingHorizontal: scale(20),
    backgroundColor: "#fff",
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 3 },
    shadowOpacity: 0.16,
    shadowRadius: 6,
    borderRadius: verticalScale(20),
    width: "100%",
    flexDirection: "row",
  },
  branchContainer: {
    flex: 2,
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "space-between",
  },
  branchImageContainer: {
    width: verticalScale(30),
    height: verticalScale(30),
    borderRadius: verticalScale(30) / 2,
    overflow: "hidden",
    alignItems: "center",
    justifyContent: "center",
  },
  branchImage: {
    width: verticalScale(40),
    height: verticalScale(40),
  },
  branchName: {
    marginTop: verticalScale(3),
    fontSize: moderateScale(6),
    fontWeight: "bold",
    color: "#6E6969",
  },
  dealInfo: {
    flex: 9,
    flexDirection: "column",
    justifyContent: "space-between",
    paddingBottom: verticalScale(6),
  },
  title: {
    fontSize: moderateScale(10),
    fontWeight: "bold",
    color: "#6E6969",
  },
  subText: {
    fontSize: moderateScale(12),
    color: "#9B9797",
  },
});
