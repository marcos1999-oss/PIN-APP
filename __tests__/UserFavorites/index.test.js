/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import UserFavorites from '../../src/Screen/UserFavorites';

const mockStore = configureMockStore();

jest.mock('react-native-cached-images', () => {
  return {
    DocumentDir: () => {},
    ImageCache: {
      get: {
        clear: () => {},
      },
    },
  }
});

test('renders correctly', () => {
  const store = mockStore({
    auth: {
      me: {}
    },
    posts: {
      data: {},
      pagination: {}
    },
    offers: {
      data: {},
    },
    firms: {
      data: {},
    },
    features: {
      data: {},
    },
    holidays: {
      data: {},
    },
    schedules: {
      data: {},
    },
    pins: {
      near: {},
    }
  });
  const tree = renderer.create(
    <Provider store={store}>
      <UserFavorites isTesting />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
