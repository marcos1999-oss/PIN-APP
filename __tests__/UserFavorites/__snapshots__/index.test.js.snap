// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#5F92F3",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingTop": 18,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          },
          Object {
            "backgroundColor": "#5F92F3",
            "borderBottomWidth": 0,
            "elevation": 0,
            "marginTop": 26.68,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            },
            Object {
              "flex": 1,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "alignSelf": "center",
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
              },
            ]
          }
          uppercase={false}
        >
          Favorites
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 50,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginRight": -9,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "alignSelf": "center",
                  "color": "white",
                  "fontSize": 18,
                  "fontWeight": "bold",
                },
                Object {
                  "fontFamily": "PinAppIcon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            î¤´
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "marginLeft": 40,
          "marginRight": 30,
        }
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      getItemLayout={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0.2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "width": "100%",
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
